docs/source/upgrade.rst

```diff
@@ -0,0 +1,747 @@
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+CuPy, which was originally a part of Chainer, has been separated into a different Python package since Chainer v2.
+It changes the way to set up Chainer with CUDA support.
+In particular, you have to separately install :mod:`cupy` package to enable CUDA support.
+See :ref:`install-guide` for the recommended installation steps.
+
+Fortunately, there is no need of updating your source code to catch up with this change.
+
+
+Global configurations
+---------------------
+
+.. _upgrade-train-mode:
+
+Training mode is configured by a thread-local flag
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In Chainer v2, the concept of *training mode* is added.
+It is represented by a thread-local flag ``chainer.config.train``, which is a part of :ref:`the unified configuration <configuration>`.
+When ``chainer.config.train`` is true, functions of Chainer run in the training mode, and otherwise they run in the test mode.
+For example, :class:`~links.BatchNormalization` and :func:`~functions.dropout` behave differently in each mode.
+
+In Chainer v1, such a behavior was configured by the ``train`` argument of each function.
+**This train argument has been removed in Chainer v2.**
+If your code is using the ``train`` argument, you have to update it.
+In most cases, what you have to do is just removing the ``train`` argument from any function calls.
+
+.. admonition:: Example
+
+   Consider the following model definition and the code to call it in test mode written for Chainer v1.
+
+   .. code-block:: py
+
+      # Chainer v1
+      import chainer.functions as F
+
+      class MyModel(chainer.Link):
+          ...
+
+          def __call__(self, x, train=True):
+              return f(F.dropout(x, train=train))
+
+      m = MyModel(...)
+      y = m(x, train=False)
+
+   In Chainer v2, it should be updated into the following code:
+
+   .. code-block:: py
+
+      # Chainer v2
+      import chainer.functions as F
+
+      class MyModel(chainer.Link):
+          ...
+
+          def __call__(self, x):
+              return f(chainer.functions.dropout(x))
+
+      m = MyModel(...)
+      with chainer.using_config('train', False):
+          y = m(x)
+
+.. _upgrade-configurations:
+
+Configurations are added and replace some of existing global flags
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+There are many global settings moved to :ref:`the unified configuration <configuration>` other than the training mode.
+Following is the complete list of the configuration entries that have corresponding features in Chainer v1.
+
+``chainer.config.debug``
+    It is corresponding to the debug mode in Chainer v1, which was configured by :func:`set_debug` and extracted by :func:`is_debug`.
+    These functions are also available in Chainer v2, so you basically do not need to update the code related to the debug mode.
+``chainer.config.deterministic``
```

Comment:
We have renamed the variable to `cudnn_deterministic` in #2710. So, please change it accordingly.